apps/user-office-frontend/src/components/proposal/ProposalAttachmentDownload.tsx

```diff
@@ -0,0 +1,327 @@
+                  {option.name}
+                </li>
+              );
+            }}
+            loading={loadingTemplates}
+          />
+        </Grid>
+
+        <Grid item xs={12}>
+          <Autocomplete
+            id="attachment-question-select"
+            aria-labelledby="attachment-question-select-label"
+            fullWidth={true}
+            noOptionsText={'No matching attachment question'}
+            disableClearable
+            multiple
+            onChange={(_event, selectedValue) => {
+              setSelectedAttachmentsQuestions(
+                selectedValue.map((value) => value.id)
+              );
+            }}
+            getOptionLabel={(option) => option.question}
+            isOptionEqualToValue={(option, value) => option.id === value.id}
+            options={attachmentQuestions.filter(
+              (question, index) =>
+                attachmentQuestions.findIndex(
+                  (value) => value.id == question.id
+                ) === index
+            )}
+            renderInput={(params) => (
+              <TextField
+                {...params}
+                key={params.id}
+                data-cy="attachmentQuestionName"
+                label="Attachment question"
+                placeholder="Select attachment question"
+              />
+            )}
+            renderOption={(props, option) => {
+              return (
+                <li {...props} key={option.id}>
+                  {option.question}
+                </li>
+              );
+            }}
+            loading={loading}
+          />
+        </Grid>
+        <Grid item xs={12}>
+          <FormControlLabel
+            control={
+              <Checkbox
+                checked={includeAllProposals}
+                onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
+                  setIncludeAllProposals(event.target.checked);
+                }}
+              />
+            }
+            label="Include proposal(s) which may not have the selected questions"
+          />
+        </Grid>
+      </Grid>
+      <Button
+        fullWidth
+        className={classes.submit}
+        disabled={
+          loadingTemplates ||
+          loading ||
+          !selectedPdfTemplate ||
+          selectedAttachmentsQuestions.length < 0
+        }
+        data-cy="proposalAttachmentDownloadButton"
+        onClick={() => {
+          const proposalIds = getProposalIds(
+            includeAllProposals,
+            proposalsData,
+            selectedAttachmentsQuestions
+          );
+          if (
+            proposalIds.length > 0 &&
+            selectedPdfTemplate &&
+            selectedAttachmentsQuestions.length > 0
+          ) {
+            const title = proposalsData.filter((proposal) =>
+              proposalIds.includes(proposal.primaryKey)
+            );
+            downloadProposalAttachment(
+              proposalIds,
+              title?.[0].title || '',
```

Comment:
This has the potentiality to throw error, as the length of the title could be 0. 

Also do we need || operator here or ??.

Reply:
What is the reasoning behind that because it achieves the same if it null or undefined make use empty string

Reply:
Okay. Regarding the potential error on `title?.[0].title`, do we have any fix?